### PR TITLE
Add "today" and "done" button.

### PR DESCRIPTION
**Pull request by knight-bubble**
https://community.openproject.org/work_packages/19905
depends on #2973

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://hirtie-maxim-fix-19905-today-done-buttons.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://hirtie-maxim-fix-19905-today-done-buttons.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by knight-bubble** [comment]
@floriank, this call isn't necessary, I just got it after rebasing. But I'll remove it.

File: frontend/tests/integration/protractor.conf.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ exports.config = {
 
   // Capabilities to be passed to the webdriver instance.
   capabilities: {
-    'browserName': 'firefox'
+    'browserName': 'chrome'
```
Comment (floriank) [inline frontend/tests/integration/protractor.conf.js]: No, no.

**No**

**Comment by floriank** [comment]
Erm, I am not quite sure what this PR is ultimately about. This seems to mix quite some issues.

My understanding was that the `jQuery.datepicker` already supported the "today" and "done" methods API-wise. The only thing imho that had to be added was a scope `$digest` in the right places. See http://api.jqueryui.com/1.10/datepicker/#option-showButtonPanel.

What exactly does this do? Is this maybe a `rebase`-failure?

File: frontend/app/misc/datepicker-defaults.js [hunk 2]
```diff
@@ -72,6 +72,16 @@ jQuery(function($) {
 
   $.extend($.datepicker, {
 
+    _gotoToday2: $.datepicker._gotoToday,
```
Comment (floriank) [inline frontend/app/misc/datepicker-defaults.js]: not a fan of this property name, maybe `originalGotoToday` or something?

**Comment by floriank** [comment]
Functionality wise this seems okay though.

**Comment by knight-bubble** [comment]
@floriank, you right for "Today" button. But when you are using inline datepicker there is no "Done".

**Comment by floriank** [comment]
What do you mean? From the API description:

> Whether to display a button pane underneath the calendar. The button pane contains two buttons, a Today button that links to the current day, and a Done button that closes the datepicker.

**Comment by knight-bubble** [comment]
@floriank, that is right but it's different for inline datepicker. http://stackoverflow.com/a/12567308

**Comment by floriank** [comment]
aha!

I did not know that.

**Comment by floriank** [comment]
I consider this merge-able, however, this depends on #2973, which currently blocks this.

**Comment by floriank** [comment]
I merged #2973 - please rebase.

**Pull request merged**